Update COC with incident handling (#652)
cncf/tag-security: CODE-OF-CONDUCT.md

     attend meetings.
   * Work performed within this group, either finalized or in draft, is to be used in accordance with the group [Mission and Charter](https://github.com/cncf/tag-security/blob/main/governance/charter.md), the open source license, and to be used for the equal benefit of all members of the community.  Further information on use of work may be found in [Security Assessments: Outcome](https://github.com/cncf/tag-security/tree/main/assessments#outcome)

# Incident handling and escalation

> If an incident that occurs that you is not in good fair, please follow the guidelines on handling and escalating the issue to TAG leadership:

Reviewer: "you is not in good fair" -- not sure what this was meant to be

> - If you see content that clearly does not meet the official Code of Conduct, please make it known to a Co-Chair/TL and the creator of the content. (For more details refer to the CNCF Code of Conduct).

Reviewer: please use new chair email list:  https://github.com/cncf/tag-security/issues/655 -- if a single chair or TL is afk, we don't want someone to get upset that they reported something and nothing was done.  Reporting must not include a single point of failure.

Author: Yup agree on no single point on failure, this PR depends on #655. (i wish github had a way to enforce this)

Reviewer 2: fair -> faith?

Author: ```suggestion
- If you see content that clearly does not meet the official Code of Conduct, please send an e-mail to the Co-Chair/TL mailing list (cncf-tag-security-leads@lists.cncf.io) and the creator of the content. (For more details refer to the CNCF Code of Conduct).
```

> - If you are uncomfortable with a piece of content (but it may not necessarily violate the COC), we suggest sending a private message to the content owner expressing your concerns. If this is not resolved, you may wish to request the help of a Co-Chair/TL to help mediate the situation.

Author: ```suggestion
- If you are uncomfortable with a piece of content (but it may not necessarily violate the COC), we suggest sending a private message to the content owner expressing your concerns. If this is not resolved, you may wish to request the help of a Co-Chair/TL via cncf-tag-security-leads@lists.cncf.io to help mediate the situation.
```

> - Discussions about these issues are important, and there are great avenues to discuss them. This includes bringing up concerns to the CNCF TOC (which can be done through discussion with TAG leadership) or talking to TAG leadership about moderating a post. To help ensure that we can give focus to these issues and not tangle them up with technical discussions, we should keep these discussions separate from channels which are focused around technical exchange.

Reviewer: TAG leadership -> Security TAG leadership (unless you meant to imply the not-widely-known group that is all the TAG chairs and might be considered to be "TAG leadership")

Author: ```suggestion
- Discussions about these issues are important, and there are great avenues to discuss them. This includes bringing up concerns to the CNCF TOC (which can be done through discussion with Security TAG leadership) or talking to Security TAG leadership about moderating a post. To help ensure that we can give focus to these issues and not tangle them up with technical discussions, we should keep these discussions separate from channels which are focused around technical exchange.
```

> - Please seek to understand, in some cases you may not agree with a decision or request, however practicing tolerance and mindfulness is just as important when this occurs

Reviewer: maybe something like.... 
- If your behavior makes someone else uncomfortable, that should be reason enough.  
- just because someone else did a thing and no one said anything, doesn't mean it was ok.

Author: Feel like this maybe goes in the top section of code of conduct? What do you think?

Author: Oh i think understand you're comment now, this is from the side of the content creator.

Author: ```suggestion
- If you receive a notice about a piece of content you've created, please seek to understand that in some cases you may not agree with a decision or request. Being able to practice tolerance and mindfulness is just as important to keep the community working towards a common goal. The mediation and resolution system that we have in place aims to handle this with the hope that both content creators and consumers are heard and represented. These situations are not zero sum, and often we aim to reach an agreeable compromise.
```

> Thanks for reading through this long post, but tl; dr, be nice, inclusive and welcoming. Misunderstandings, mistakes and oversights happen, and when they do, there are some good ways to go about addressing them with the TAG and CNCF!

Reviewer: Can you rearrange this sentence to not include acronyms (which are not super-inclusive to newcomers)?  I think you could just drop "with the TAG and CNCF" though then the sentence doesn't feel as friendly.   Maybe something simple like "good ways to go about collaborating with colleagues to make our community inclusive and welcoming to everyone!"

Also I love "TL; DR," but lots of folks in different communities and English-as-second-language may not know the acronym.

Author: ```suggestion
In summary, be nice, inclusive and welcoming. Misunderstandings, mistakes and oversights happen, and when they do, there are some good ways to go about having a conversation with colleagues to make our community inclusive and welcoming to everyone!
```

For content creators:
- If you receive a notice about a piece of content you've created, please seek to understand that in some cases you may not agree with a decision or request. Being able to practice tolerance and mindfulness is just as important to keep the community working towards a common goal. The mediation and resolution system that we have in place aims to handle this with the hope that both content creators and consumers are heard and represented. These situations are not zero sum, and often we aim to reach an agreeable compromise where a discussion of a topic can happen without making members of the community feel uncomfortable.
- In the event where there is disagreement, we have some guidelines that can help prevent escalation
  - Do not rationalize the actions you take. We do not expect anyone to understand what everyone else feels towards certain things (e.g. the same gestures in certain cultures are good and bad in others). Understand that something may not be wrong, but it may affect others.
  - Do not bring the discussion out of context.

> In summary, be nice, inclusive and welcoming. Misunderstandings, mistakes and oversights happen, and when they do, there are some good ways to go about having a conversation with colleagues to make our community inclusive and welcoming to everyone!

Author: ```suggestion

In summary, be nice, inclusive and welcoming. Misunderstandings, mistakes and oversights happen, and when they do, there are some good ways to go about having a conversation with colleagues to make our community inclusive and welcoming to everyone!
```

     attend meetings.
   * Work performed within this group, either finalized or in draft, is to be used in accordance with the group [Mission and Charter](https://github.com/cncf/tag-security/blob/main/governance/charter.md), the open source license, and to be used for the equal benefit of all members of the community.  Further information on use of work may be found in [Security Assessments: Outcome](https://github.com/cncf/tag-security/tree/main/assessments#outcome)

# Incident handling and escalation

If an incident involving community conduct occurs, please follow the guidelines on handling and escalating the issue to TAG leadership:

> - If you see content that clearly does not meet the official Code of Conduct, please send an e-mail to the Co-Chair/TL mailing list (cncf-tag-security-leads@lists.cncf.io) and the creator of the content. (For more details refer to the CNCF Code of Conduct).

Reviewer: pls add link to CNCF Code of Conduct

Reviewer: I also think we need something in the preamble that defines "content" as not only published content, also online discourse, such as slack messages or emails, as well as interactions at in-person events.

Reviewer 3: ```suggestion
- If you see content that clearly does not meet the official Code of Conduct, please send an e-mail to the Co-Chair/TL mailing list (cncf-tag-security-leads@lists.cncf.io) and the creator of the content. (For more details refer to the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md)).
```

For content creators:
- If you receive a notice about a piece of content you've created, please seek to understand that in some cases you may not agree with a decision or request. Being able to practice tolerance and mindfulness is just as important to keep the community working towards a common goal. The mediation and resolution system that we have in place aims to handle this with the hope that both content creators and consumers are heard and represented. These situations are not zero sum, and often we aim to reach an agreeable compromise where a discussion of a topic can happen without making members of the community feel uncomfortable.
- In the event where there is disagreement, we have some guidelines that can help prevent escalation

>   - Do not rationalize the actions you take. We do not expect anyone to understand what everyone else feels towards certain things (e.g. the same gestures in certain cultures are good and bad in others). Understand that something may not be wrong, but it may affect others.
>   - Do not bring the discussion out of context.

Author: ```suggestion
  - Do not bring the discussion out of context.
  - Do not rationalize the actions you take. We do not expect anyone to understand what everyone else feels towards certain things (e.g. the same gestures in certain cultures are good and bad in others). Understand that something may not be wrong, but it may affect others.
```

     in [Security Reviews:
     Outcome][review-outcome]

# Incident handling and escalation

> Content for the purposes of the code of conduct as well as incident is defined not only as published or draft content but also online discourse, such as slack messages or emails, and interactions at in-person events

Author: ```suggestion
Content for the purposes of the code of conduct as well as incident is defined not only as published or draft content but also online discourse, such as slack messages or emails, and interactions at in-person events.
```

If an incident involving community conduct occurs, please follow the guidelines below on how to handle and report the issue:

> - If you see content that clearly does not meet the official Code of Conduct, please send an e-mail to the Co-Chair/TL mailing list (cncf-tag-security-leads@lists.cncf.io) and the creator of the content. (For more details refer to the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md)).

Author: ```suggestion
- If you see content that clearly does not meet the official Code of Conduct, please send an e-mail to the Co-Chair/TL mailing list (cncf-tag-security-leads@lists.cncf.io) and the creator of the content. (For more details refer to the [CNCF Code of Conduct][cncf-coc]).
```

> - If you are uncomfortable with a piece of content (but it may not necessarily violate the COC), we suggest sending a private message to the content owner expressing your concerns. If this is not resolved, you may wish to request the help of a Co-Chair/TL via cncf-tag-security-leads@lists.cncf.io to help mediate the situation.

Author: ```suggestion
- If you are uncomfortable with a piece of content (but it may not necessarily violate the code of conduct), we suggest sending a private message to the content owner expressing your concerns. If this is not resolved, you may wish to request the help of a Co-Chair/TL via cncf-tag-security-leads@lists.cncf.io to help mediate the situation.
```

> - Discussions about these issues are important, and there are great avenues to discuss them. This includes bringing up concerns to the [CNCF TOC](https://www.cncf.io/people/technical-oversight-committee/) (which can be done through discussion with Security TAG leadership) or talking to Security TAG leadership about moderating a post. To help ensure that we can give focus to these issues and not tangle them up with technical discussions, we should keep these discussions separate from channels which are focused around technical exchange.

Author: ```suggestion
- Discussions about these issues are important, and there are great avenues to discuss them. This includes bringing up concerns to the [CNCF TOC][cncf-toc] (which can be done through discussion with Security TAG leadership) or talking to Security TAG leadership about moderating a post. To help ensure that we can give focus to these issues and not tangle them up with technical discussions, we should keep these discussions separate from channels which are focused around technical exchange.
```

For content creators:

> - Content must strive to remain _on-topic_, particularly where video and images are provided.  Use of emojis and gifs as responses are content in and of themselves need to be relevant to the particular post. For examples please refer the the reference area below. 

Author: ```suggestion
- Content must strive to remain _on-topic_, particularly where video and images are provided.  Use of emojis and gifs as responses are content in and of themselves need to be relevant to the particular post. For examples please refer to the reference section below. 
```

     in [Security Reviews:
     Outcome][review-outcome]

# Incident handling and escalation

Content for the purposes of the code of conduct as well as incident is defined
not only as published or draft content but also online discourse, such as slack
messages or emails, and interactions at in-person events.  If an incident
involving community conduct occurs, please follow the guidelines below on how to
handle and report the issue:

- If you see content that clearly does not meet the official Code of Conduct,
  please send an e-mail to the Co-Chair/TL mailing list
  (cncf-tag-security-leads@lists.cncf.io) and the creator of the content. (For
  more details refer to the [CNCF Code of Conduct][cncf-coc]).
- If you are uncomfortable with a piece of content (but it may not necessarily
  violate the code of conduct), we suggest sending a private message to the
  content owner expressing your concerns. If this is not resolved, you may wish
  to request the help of a Co-Chair/TL via cncf-tag-security-leads@lists.cncf.io
  to help mediate the situation.

> - Discussions about these issues are important, and there are great avenues to

Reviewer 3: I think `these issues` should be expanded to `potential code of conduct violations and concerns`

> - Discussions about these issues are important, and there are great avenues to

Author: ```suggestion
- Discussions about these potential code of conduct violations and concerns are important, and there are great avenues to
```

  not zero sum, and often we aim to reach an agreeable compromise where a
  discussion of a topic can happen without making members of the community feel
  uncomfortable.
- In the event where there is disagreement, we have some guidelines that can
  help prevent escalation
  - Do not bring the discussion out of context.
  - Do not rationalize the actions you take. We do not expect anyone to
    understand what everyone else feels towards certain things (e.g. the same
    gestures in certain cultures are good and bad in others). Understand that
    something may not be wrong, but it may affect others.

In summary, be nice, inclusive and welcoming. Misunderstandings, mistakes and
oversights happen, and when they do, there are some good ways to go about having
a conversation with colleagues to make our community inclusive and welcoming to
everyone!  

## Reference

Example of reasonable gif: Group is close to wrapping up deliverable, as part of
an update, the lead posts a "nearly done" gif.  

Example of reasonable emoji:  Post in the group uses emojis to break up content
and is relevant to the item discussed or used in response to post to signify

> voting, opinion, acceptance, emotion, etc.  

Author: I'm not sure if i know how to describe it. I don't think ive personally seen a case of unacceptable emojis being used, but im not a big GIF/emoji user, so not too familiar with what it would constitute.. i do think stating what's ok would be useful enough for folks to be able to vet their own content.

> voting, opinion, acceptance, emotion, etc.  

Reviewer 3: @ashutosh-narkar i think for now this is sufficient.  If/when an unacceptable gif or emoji comes up, we can enumerate it here.

> voting, opinion, acceptance, emotion, etc.  

Reviewer 4: Sounds good.

Example of reasonable image and video:  Posting a picture of a community meetup*
or posting a recording to a presentation on cloud native security.

*Note: Many events within the community may show alcohol consumption, it is
important that when posting photos and videos members consider if the post
glorifies alcohol or alcohol is the primary subject of the content
(unacceptable) or if the alcohol is happenstance occurrence in the image
(acceptable).  

> 

Reviewer 4: IMO this is very subjective. I mean who's to say if a video "glorifies alcohol" or not. So are we saying we're going to slap a warning label on the content or remove it altogether ?  Also wouldn't this need to be extended for drugs etc. I'm curious if any other org has similar guidelines.

Reviewer 5: I agree Ashutosh, we need to extend it to number of things actually, including Drugs

Author: ```suggestion
*Note: Many events within the community may include content which is only acceptable depending on the context it is used in.
An example of this is alcohol consumption. It is
important that when posting photos and videos members consider if the post
glorifies alcohol or alcohol is the primary subject of the content
(unacceptable) or if the alcohol is happenstance occurrence in the image
(acceptable).  

```

Changed it to be an example instead.

Reviewer 3: A lot of this is very subjective.  We can provide some examples but realistically so long as everyone is not raised the same way this will continue to vary widely.  Example: drug use is not the same globally, some places items are classified as drugs and illegal or legal and in others they are not.

     in [Security Reviews:
     Outcome][review-outcome]

# Incident handling and escalation

Content for the purposes of the code of conduct as well as incident is defined
not only as published or draft content but also online discourse, such as slack
messages or emails, and interactions at in-person events.  If an incident
involving community conduct occurs, please follow the guidelines below on how to
handle and report the issue:

- If you see content that clearly does not meet the official Code of Conduct,

>   please send an e-mail to the Co-Chair/TL mailing list

Reviewer 6: Consider including some mechanism in case the incident is regarding the behavior of a co-chair?

  (cncf-tag-security-leads@lists.cncf.io) and the creator of the content. (For

>   more details refer to the [CNCF Code of Conduct][cncf-coc]).

Reviewer 3: ```suggestion
  more details refer to the [CNCF Code of Conduct][cncf-coc]).  If it is regarding a 
  co-chair, reach out to the two other chairs directly if you are uncomfortable using
  the mailing list.
```

     in [Security Assessments:
     Outcome](https://github.com/cncf/tag-security/tree/main/assessments#outcome)

# Incident handling and escalation

> 

Reviewer 3: ```suggestion

Content for the purposes of the code of conduct as well as incident is defined not only as published or draft content but also online discourse, such as slack messages or emails, and interactions at in-person events
```

Author: add "."

> If an incident involving community conduct occurs, please follow the guidelines on handling and escalating the issue to TAG leadership:

Reviewer: ```suggestion
If an incident involving community conduct occurs, please follow the guidelines below on how to handle and report the issue:
```

> - If you see content that clearly does not meet the official Code of Conduct, please send an e-mail to the Co-Chair/TL mailing list (cncf-tag-security-leads@lists.cncf.io) and the creator of the content. (For more details refer to the CNCF Code of Conduct).

Reviewer 3: ```suggestion
- If you see content that clearly does not meet the official Code of Conduct, please send an e-mail to the Co-Chair/TL mailing list (cncf-tag-security-leads@lists.cncf.io) and the creator of the content. (For more details refer to the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/master/code-of-conduct.md)).
```

> - If you are uncomfortable with a piece of content (but it may not necessarily violate the COC), we suggest sending a private message to the content owner expressing your concerns. If this is not resolved, you may wish to request the help of a Co-Chair/TL via cncf-tag-security-leads@lists.cncf.io to help mediate the situation.

Reviewer: COC needs to defined at the top of the document, probably in the title (will also set acronym standards, for example, I usually use CoC, but don't actually care as long as it's consistent.

> - Discussions about these issues are important, and there are great avenues to discuss them. This includes bringing up concerns to the CNCF TOC (which can be done through discussion with Security TAG leadership) or talking to Security TAG leadership about moderating a post. To help ensure that we can give focus to these issues and not tangle them up with technical discussions, we should keep these discussions separate from channels which are focused around technical exchange.

Reviewer: newcomers to our group won't know what the TOC is, maybe link here?  https://www.cncf.io/people/technical-oversight-committee/

Reviewer 3: ```suggestion
- Discussions about these issues are important, and there are great avenues to discuss them. This includes bringing up concerns to the [CNCF TOC](https://www.cncf.io/people/technical-oversight-committee/) (which can be done through discussion with Security TAG leadership) or talking to Security TAG leadership about moderating a post. To help ensure that we can give focus to these issues and not tangle them up with technical discussions, we should keep these discussions separate from channels which are focused around technical exchange.
```

For content creators:

> - If you receive a notice about a piece of content you've created, please seek to understand that in some cases you may not agree with a decision or request. Being able to practice tolerance and mindfulness is just as important to keep the community working towards a common goal. The mediation and resolution system that we have in place aims to handle this with the hope that both content creators and consumers are heard and represented. These situations are not zero sum, and often we aim to reach an agreeable compromise where a discussion of a topic can happen without making members of the community feel uncomfortable.

Reviewer 3: ```suggestion

- Content must strive to remain _on-topic_, particularly where video and images are provided.  Use of emojis and gifs as responses are content in and of themselves need to be relevant to the particular post. For examples please refer the the reference area below. 
- If you receive a notice about a piece of content you've created, please seek to understand that in some cases you may not agree with a decision or request. Being able to practice tolerance and mindfulness is just as important to keep the community working towards a common goal. The mediation and resolution system that we have in place aims to handle this with the hope that both content creators and consumers are heard and represented. These situations are not zero sum, and often we aim to reach an agreeable compromise where a discussion of a topic can happen without making members of the community feel uncomfortable.
```

Author: "refer to the"

- In the event where there is disagreement, we have some guidelines that can help prevent escalation
  - Do not bring the discussion out of context.
  - Do not rationalize the actions you take. We do not expect anyone to understand what everyone else feels towards certain things (e.g. the same gestures in certain cultures are good and bad in others). Understand that something may not be wrong, but it may affect others.

In summary, be nice, inclusive and welcoming. Misunderstandings, mistakes and oversights happen, and when they do, there are some good ways to go about having a conversation with colleagues to make our community inclusive and welcoming to everyone!

> 

Reviewer 3: ```suggestion
## Reference

Example of reasonable gif: Group is close to wrapping up deliverable, as part of an update, the lead posts a "nearly done" gif.  

Example of reasonable emoji:  Post in the group uses emojis to break up content and is relevant to the item discussed or used in response to post to signify voting, opinion, acceptance, emotion, etc.  

Example of reasonable image and video:  Posting a picture of a community meetup* or posting a recording to a presentation on cloud native security.

*Note: Many events within the community may show alcohol consumption, it is important that when posting photos and videos members consider if the post glorifies alcohol or alcohol is the primary subject of the content (unacceptable) or if the alcohol is happenstance occurrence in the image (acceptable).
```

[cncf-coc]: https://github.com/cncf/foundation/blob/master/code-of-conduct.md  
[charter]: https://github.com/cncf/tag-security/blob/main/governance/charter.md

> [review-outcome]: https://github.com/cncf/tag-security/tree/main/assessments#outcome

Author: ```suggestion
[review-outcome]: https://github.com/cncf/tag-security/tree/main/assessments#outcome
[cncf-toc]: https://www.cncf.io/people/technical-oversight-committee/
```
